pyspedas/utilities/download.py

```diff
@@ -399,12 +530,30 @@ def download(
 
         short_path = local_file[: 1 + local_file.rfind("/")]
 
+        if is_fsspec_uri(url) and no_download:
+            # this is the case where a user wants to download to local
+            # from a URI (NOT RECOMMENDED)
+            #print('NOT RECOMMENDED!')
+            protocol, path = url.split("://")
+            fs = fsspec.filesystem(protocol, anon=False)
+
+            links = [link[link.rfind("/")+1:] for link in fs.glob(url)]
```

Comment:
Maybe add a note on why you have to do this. To check my understanding this is because on s3 or cloud this isn't a real path so we have to check every file that contains the "path"?